sycl/include/CL/sycl/buffer.hpp

```diff
@@ -126,6 +139,26 @@ class buffer {
         detail::getNextPowerOfTwo(sizeof(T)), propList);
   }
 
+  // This constructor is a prototype for a future SYCL specification
+  template <class Container, int N = dimensions,
+            typename = EnableIfOneDimension<N>,
+            typename = EnableIfContiguous<Container>>
```

Comment:
I don't think there is a good reason to require the contiguous container concept here. I think we should only require the range concept here (whether it is has a begin and end) and then only use contiguous container property (if present) for the optimization that no memcpy is needed in some cases.

Reply:
> For now we only support write-back to contiguous storage. 

Maybe I'm missing something, but we support write back to non const non contiguous containers, Check set_final_data method in sycl_mem_obj_t.hpp:178   .

Reply:
Yes I was wrong. We do support it. I still think we should wait to add support for non-contiguous containers until we have resolved how exactly write-back is supposed to be decided. Is that OK?